gammapy/astro/darkmatter/utils.py

```diff
@@ -52,3 +66,181 @@ def compute_jfactor(self):
+    annihilation simulated dataset is fitted to a list of `~gammapy.astro.darkmatter.DMAnnihilation`
+    models. These are created within the range of the given lists of annihilation channels and particle
+    masses. For each fit, the value of the scale parameter that makes :math:`\Delta TS > 2.71` is
+    multiplied by the thermal relic cross section, and subsequently taken as the estimated value of
+    :math:`\sigma\nu`.
+
+    Parameters
+    ----------
+    dataset : `~gammapy.cube.fit.MapDataset`
+        Simulated dark matter annihilation dataset.
+    masses : list of `~astropy.units.Quantity`
+        List of particle masses where the values of :math:`\sigma\nu` will be calculated.
+    channels : list of strings allowed in `~gammapy.astro.darkmatter.PrimaryFlux`
+        List of channels where the values of :math:`\sigma\nu` will be calculated.
+    jfact : `~astropy.units.Quantity` (optional)
+        Integrated J-Factor needed when `~gammapy.image.models.SkyPointSource` spatial model is used, default value 1.
+    absorption_model : `~gammapy.spectrum.models.Absorption` (optional)
+        Absorption model, default is None.
+    z: float (optional)
+        Redshift value, default value 0.
+    k: int (optional)
+        Type of dark matter particle (k:2 Majorana, k:4 Dirac), default value 2.
+    xsection: `~astropy.units.Quantity` (optional)
+        Thermally averaged annihilation cross-section, default value declared in `~gammapy.astro.darkmatter.DMAnnihilation`.
+
+    Examples
+    --------
+    This is how you may run the `SigmaVEstimator`::
+
+        import logging
+        logging.basicConfig()
+        logging.getLogger("gammapy.astro.darkmatter.utils").setLevel("INFO")
+
+        jfact = 3.41e19 * u.Unit("GeV2 cm-5")
+        channels = ["b", "t", "Z"]
+        masses = [70, 200, 500, 5000, 10000, 50000, 100000]*u.GeV
+        estimator = SigmaVEstimator(simulated_dataset, masses, channels, jfact=JFAC)
+        result = estimator.run()
+    """
+
+    def __init__(
+        self,
+        dataset,
+        masses,
+        channels,
+        jfact=1,
+        absorption_model=None,
+        z=0,
+        k=2,
+        xsection=None,
+    ):
+
+        self.dataset = dataset
+        self.masses = masses
+        self.channels = channels
+        self.jfact = jfact
+        self.absorption_model = absorption_model
+        self.z = z
+        self.k = k
+
+        if not xsection:
+            xsection = DMAnnihilation.THERMAL_RELIC_CROSS_SECTION
+        self.xsection = xsection
+
+        self._spatial_model = dataset.model.spatial_model
+        self._geom = dataset.counts.geom
+        self._exposure = dataset.exposure
+        self._background_model = dataset.background_model
+        self._psf = dataset.psf
+        self._edisp = dataset.edisp
+
+        self._counts_map = WcsNDMap(self._geom, np.random.poisson(dataset.npred().data))
```

Comment:
I think the simulated dataset already contains a simulated counts map, so maybe remove?

Reply:
I had imagined calling this `SigmaVEstimator.run()` inside an external loop of n runs (see cell[13] in notebook https://github.com/peroju/dm-gammapy/blob/master/notebooks/DarkMatterUseCaseUpgraded.ipynb and the TODO comment in the code cell) Every call in the loop should create a different random poisson counts map from our simulated dataset, and the results of each individual fitting of the run would be averaged at the end.

I had thought of providing a different tool later (if needed) building this loop of n runs, using `SigmaVEstimator.run()` internally in the loop and throwing the averaged results. 

Does this make sense?